doc/arch/adf-006-register-serialization-format.md

```diff
@@ -0,0 +1,101 @@
+# Decision record: Register Serialization Format
+
+## Context ##
+
+The basic components of a Register are Items and Entries. The Items are the content of the Register and are currently expressed as JSON objects e.g.
+
+    {
+      "address": "5T8EXT7N",
+      "denominations": [
+        "C22"
+      ],
+      "dioceses": [
+        "CE23"
+      ],
+      "headteacher": "Mr T Wilson",
+      .....
+    }
+
+An Entry refers to an Item by the Item's SHA-256 hash and also contains a timestamp recording when the Item was added to the Register.
+
+The algorithm for computing the hash involves canonicalizing the JSON by removing all whitespace between elements i.e. not in a quoted string, calculating the SHA-256 hash on the JSON encoded as a UTF-8 string and expressed as a Hex string.
+
+The other key feature of a Register is its Root Hash which is the root element of the Merkle tree based on the sequence of Entries. The Root Hash can be used in combination with a Proof composed of a number of other hash values to verify a single Entry.
+
+Register data is currently provided by the Register Design Authority usually in TSV format e.g.
+
+    402378[tab]Abertillery Learning Community[tab][tab]677[tab][tab][tab]  [tab][tab][tab][tab]30[tab][tab][tab][tab][tab][tab]2016-09-01[tab]
+
+The TSV does not fully represent a Register in that the timestamps of the Entries and the Root Hash are not included.
+
+A Register is not the same thing as the Openregister-java application. It can be represented in an external, human readable format making the structure of the Register explicit. We call this the Register Serialization Format (RSF).
+
+The RSF can also be used to clone an existing Register as a whole or in part.
+
+Registers can be large - the address register currently has c34 million Entries and the food-premises-rating register c500,000. So the RSF should not need to be parsed as a whole before processing.
+
+### Validation
+
+To be valid:
+
+- All Items should be referred to by at least one Entry. There should be no Orphan Items.
+- All Entries should refer to an Item. In theory >1 Entry can point to the same Item. Where the RSF represents part of a Register, the Item may be part of the existing Register.
+
```

Comment:
Can you emphasise that id the item is already part of the register then it does not need to be part of the RSF file?